components/search/SearchProfiles.vue

```diff
@@ -0,0 +1,71 @@
+<template>
+  <div v-if="isLoading">
+    <SearchResultItem v-for="item in 5" :key="item" is-loading />
+  </div>
+  <div v-else-if="!profileSuggestion.length" class="mx-6 mt-4">
+    {{ $t('search.profileNotFound', [name]) }}
+  </div>
+  <div v-else>
+    <div
+      v-for="item in profileSuggestion"
+      :key="item.address"
+      :value="item"
+      class="link-item"
+      @click="openProfilePage(item.address)">
+      <SearchResultItem :image="item.image">
+        <template #content>
+          <div class="flex flex-row justify-between pt-2 pr-2">
+            <span
+              class="font-bold max-w-[34ch] overflow-hidden text-ellipsis whitespace-nowrap"
+              >{{ item.name }}</span
+            >
+          </div>
+          <div class="flex flex-row justify-between pr-2">
+            <span
+              class="max-w-[34ch] overflow-hidden text-ellipsis whitespace-nowrap"
+              >{{ item?.description }}</span
+            >
+          </div>
+        </template>
+      </SearchResultItem>
+    </div>
+  </div>
+</template>
+
+<script setup lang="ts">
+import { searchProfiles } from '@/services/profile'
+import type { Profile } from '@/services/profile'
+import { useQuery } from '@tanstack/vue-query'
+import { isEthereumAddress } from '@polkadot/util-crypto'
+
+const props = defineProps({
+  name: {
+    type: String,
+    required: true,
+  },
+})
+
+const router = useRouter()
+const { urlPrefix } = usePrefix()
+const { isEvm } = useIsChain(urlPrefix)
+
+const emit = defineEmits(['close'])
+
+const { data: result, isLoading } = useQuery<{ data: Profile[] } | null>({
+  queryKey: ['search-profiles', computed(() => props.name)],
+  queryFn: () => (props.name ? searchProfiles(props.name, 10) : null),
+})
+
+const profileSuggestion = computed(() => result.value?.data || [])
```

Comment:
let's make it `profileSuggestion` plural as is an array

would it be cleaner if we used this in `useQuery`? 

![CleanShot 2024-07-09 at 09 47 40@2x](https://github.com/kodadot/nft-gallery/assets/44554284/7c9ad2f8-5bc2-436e-83c0-6d424eeef3e5)

Reply:
There is a type issue here so I use another workaround instead